fault/tester/ready_valid_tester.py

```diff
@@ -0,0 +1,52 @@
+import magma as m
+from fault.tester.staged_tester import Tester
+from fault.assert_immediate import assert_immediate
+
+
+class ReadyValidTester:
+    def __init__(self, circuit: m.Circuit, sequences: dict):
+        class Wrapper(m.Circuit):
+            io = m.IO(done=m.Out(m.Bit))
+            inst = circuit()
+            done = m.Bit(1)
+            for key, value in sequences.items():
+                port = getattr(inst, key)
+                n = len(value)
+                # Counter to advance through sequence
+                count = m.Register(m.UInt[(n - 1).bit_length()],
+                                   has_enable=True)()
+                count.I @= count.O + 1
+
+                # Current sequence elem chosen by counter
+                curr = m.mux(value, count.O)
```

Comment:
Isn't the idea to use a fifo here?

Reply:
I'm not sure what a FIFO would add to this? We'd still need to drive/monitor the input/output sequences with this logic, we could stick a fifo in between the drivers/monitors but not sure how that would help (maybe useful for adding random stall logic)

Reply:
i.e. the FIFO has a ready/valid interface, so rather than plugging into the interface directly, we plug into the fifo in between them, not sure what this adds besides maybe allowing us to fill the fifo while the DUT is stalled?